enterprise/api/v1/views.py

```diff
@@ -193,11 +193,18 @@ def get_serializer_class(self):
     def basic_list(self, request, *arg, **kwargs):
         """
         Enterprise Customer's Basic data list without pagination
+
+        Two query parameters are supported:
+        - name_or_uuid: filter by name or uuid in a single query parameter
+        - startswith: filter by name starting with the given string
         """
         startswith = request.GET.get('startswith')
+        name_or_uuid = request.GET.get('name_or_uuid')
         queryset = self.get_queryset().order_by('name')
         if startswith:
             queryset = queryset.filter(name__istartswith=startswith)
+        if name_or_uuid:
+            queryset = queryset.filter(Q(name__icontains=name_or_uuid) | Q(uuid__icontains=name_or_uuid))
```

Comment:
I forgot to ask: are you intending these to be substring matches or exact matches?  substring matching for UUIDs never really makes sense, so I feel like at least the uuid_icontains needs to be changed to uuid=uuid.UUID(name_or_uuid), but that would throw an exception, so this line would need to be re-architected:
```suggestion
        if name_or_uuid:
            try:
                requested_customer uuid = uuid.UUID(name_or_uuid)
                queryset = queryset.filter(uuid=requested_customer_uuid)
            except ValueError:
                queryset = queryset.filter(name__icontains=name_or_uuid)
```

Reply:
Huddled with Hamzah to answer my question: This query param actually needs to be a substring search on BOTH name and uuid because it's used in the frontend to drive autocompletion.

So, I'm happy with the current implementation, but I'd recommend additional details in the docstring to explain what the field is used for by the caller, and that under the hood it is a substring search, not an exact matcher.